include/ht_file_interface.h

```diff
@@ -14,49 +14,200 @@
 
 #pragma once
 
-#include <ht_platform.h>
-#include <ht_string.h>
-#include <ht_os.h>
-#include <ht_noncopy.h>
+#include <ht_platform.h> //HT_API
+#include <string> //std::string typedef
+#include <cstddef> //size_t typedef
+#include <cstdio> //FILE typedef
+#include <ht_noncopy.h> //INonCopy
 
 namespace Hatchit {
 
     namespace Core {
 
-        enum class FileSeek
-        {
-            Current,
-            End,
-            Set
-        };
-
-        enum class FileMode
-        {
-            ReadText,
-            ReadBinary,
-            WriteText,
-            WriteBinary,
-            AppendText,
-            AppendBinary
-        };
-
+        /**
+        \interface IFile
+        \ingroup HatchitCore
+        \brief An interface to describe system files.
+        
+        Interface describes the common functionality of files.
+        
+        **/
         class HT_API IFile : public INonCopy
```

Comment:
I believe we didn't want interfaces to inherit from each other, so should we move this inheritance up to the File class?